src/main/java/com/based/basedsurvey/controller/FillSurveyController.java

```diff
@@ -0,0 +1,100 @@
+package com.based.basedsurvey.controller;
+import com.based.basedsurvey.data.*;
+import com.based.basedsurvey.repo.QuestionRepository;
+import com.based.basedsurvey.repo.SurveyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import java.util.List;
+
+@Controller
+public class FillSurveyController {
+
+    SurveyRepository surveyRepository;
+
+    @Autowired
+    public FillSurveyController(SurveyRepository surveyRepository){
+        this.surveyRepository = surveyRepository;
+    }
+
+    @GetMapping("/survey/{surveyID}/answer")
+    public String getAnswer(@PathVariable String surveyID, Model model) {
+        model.addAttribute("surveyID", surveyID);
+        Survey s = surveyRepository.findSurveyById(Long.parseLong(surveyID));
+
+        model.addAttribute("surveyName", s.getName());
+
+
+        if(s.getQuestions().size() == 0){
+            model.addAttribute("empty_or_not", true);
+            return "Answer";
+        }
+
+        // formInputs will contain a dynamically generated string with all the html form inputs
+        String formInputs = "";
+
+        // for each question in survey
+        for(int index = 0; index < s.getQuestions().size(); index+=1){
+            Question q = s.getQuestions().get(index);
+
+            // case 1, the question is a multiple choice question
```

Comment:
You should make private methods or better yet, a helper class for to deal with these cases